spec/services/random_phrase_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'Random Phrase' do 
+    describe 'Happy Path' do 
+        context 'insult', :vcr do 
+            before do 
+                @response = RandomPhraseService.generate_phrase('insults')
+            end
+
+            it 'can return an insult response' do 
+                expect(@response).to be_a Hash
+                expect(@response.keys).to eq [:data]
+
+                expect(@response[:data]).to be_a Hash
+                expect(@response[:data].keys).to eq [:id, :type, :attributes]
+
+                expect(@response[:data][:id]).to be_a String
+                expect(@response[:data][:id]).to eq '1'
+
+                expect(@response[:data][:type]).to be_a String
+                expect(@response[:data][:type]).to eq 'insult'
+
+                expect(@response[:data][:attributes]).to be_a Hash
+                expect(@response[:data][:attributes].keys).to eq [:insult_phrase]
+                expect(@response[:data][:attributes].values).to eq ["You're about as useful as a screen door on a submarine."]
```

Comment:
from my understanding, when testing the Service we are more concerned about the datatypes we are recieving than the actual values. 

here we may be better off testing that the @response will have [:data][:attributes] that is a String, rather than what the string contains. if we know to expect a String, we will know how we need to handle it.

that being said, i inferred that the value will be a String based on the response... but can i be 100% sure?
if i saw _expect(@response[:data][:attributes]).to be_a(String)_  i would be 100% sure

Reply:
We can test for the actual response (ex: "You're about as useful as a screen door on a submarine." ) if we stub in the response as test data in our spec file. But you are correct Sergio, we should be more concerned with the data type ("string") than the actual data returned in this instance.